app/jinja_filters.py

```diff
@@ -584,12 +585,23 @@ def map_list_collector_config(
                     "text": remove_link_text,
                     "ariaLabel": remove_link_aria_label.format(item_name=item_name),
                     "url": list_item.get("remove_link"),
-                    "attributes": {"data-qa": "remove-item-link"},
+                    "attributes": {"data-qa": f"remove-item-link-{item_index}"},
                 }
             )
 
         rows.append(
-            {"rowTitle": item_name, "rowItems": [{"icon": icon, "actions": actions}]}
+            {
+                "rowTitle": item_name,
+                "rowItems": [
+                    {
+                        "icon": icon,
+                        "actions": actions,
+                        "rowTitleAttributes": {
```

Comment:
I did originally change `rowTitleAttributes` to be the same level as `rowTitle` but that broke some functionality because the data-qa being set in `rowTitleAttributes` was being passed `rowItems` and not `row`. If it is at the same level as `rowTitle` it doesn't have access to it. Maybe it should be renamed? or we would have to do a refactor so that it is able to deal with the whole row inside the bit where the data-qa gets set?

Reply:
Should `rowTitle` be set at the same level as `rowTitleAttributes` here? Looking at the design system template, it seems to allow for that.

Reply:
Yeah just checked, fair enough, ill do that then